docs/excel/custom-functions-troubleshooting.md

```diff
@@ -0,0 +1,65 @@
+
+## Enable runtime logging
+
+If you are testing your add-in in Office on Windows, you should [enable runtime logging](https://docs.microsoft.com/en-us/office/dev/add-ins/testing/troubleshoot-manifest#use-runtime-logging-to-debug-your-add-in) to troubleshoot issues with your add-in's XML manifest file, as well as several installation and runtime conditions. Runtime logging writes `console.log` statements to a log file to help you uncover issues. For more information about runtime logging, see [Use runtime logging to debug your add-in](https://docs.microsoft.com/en-us/office/dev/add-ins/testing/troubleshoot-manifest#use-runtime-logging-to-debug-your-add-in).  
+
+### Check for Excel error messages
+
+Excel has a number of built-in error messages which are returned to a cell if there is calculation error. These are `#NULL!`, `#DIV/0!`, `#VALUE!`, `#REF!`, `#NAME?`, `#NUM!`, `#N/A`, and `#GETTING_DATA`. At this time, custom functions can only use the error codes listed above when custom functions hit an error.
+
+## Common issues
+
+### Verify SSL certificates
+
+If your add-in fails to register, [verify that the SSL certificates are configured correctly](https://github.com/OfficeDev/generator-office/blob/master/src/docs/ssl.md) for the web server that's hosting your add-in application. Typically if you forget to do this step, you will see an error message in Excel warning you that custom functions could not be installed properly. For more information on this verification, see [Adding self-signed certificates as trusted root certificate](https://github.com/OfficeDev/generator-office/blob/master/src/docs/ssl.md).
+
+### Associate your functions
+
+In your custom functions' script file, you need to both write the function code and code to associate the function's name to the corresponding JSON id. Typically this declaration is made after each function or at the end of the script file. It's common to forget to include this and can be a reason why your functions don't work.
+
+The following example shows an add function, followed by the function's name `add` being associated with the corresponding JSON id `ADD`.
+
+```js
+function add(first, second){
+  return first + second;
+}
+
+CustomFunctions.associate("ADD", add);
+```
+
+For more information on this process, see [Associating function names with json metadata](https://docs.microsoft.com/en-us/office/dev/add-ins/excel/custom-functions-best-practices#associating-function-names-with-json-metadata).
+
+### Ensure promises return
+
+In addition to the ordinary reasons that a cell might report a #GETTING_DATA error, custom functions also will report `#GETTING_DATA` if a promise does not return a result. Check your functions to make sure that any promises are properly returning a result to a cell.
```

Comment:
```suggestion
When Excel is waiting for a custom function to complete, it displays #GETTING_DATA in the cell. If your custom function code returns a promise, but the promise does not return a result, Excel will continue showing #GETTING_DATA. Check your functions to make sure that any promises are properly returning a result to a cell.
```
I figured we should just go ahead and explain why you see GETTING_DATA (but are there other reasons? I just have the one.)

Reply:
Oh also we have to localize GETTING_DATA and other cell errors, so I left off the tick marks.

Reply:
There aren't other reasons I know of offhand, but I have seen it occur as a weird problem that crops up (kind of like #VALUE in current build), which is why I'm vague here. However, probably best to assume that it won't crop up if it isn't currently doing so